### PR TITLE
Add Kustomization files for rsyslog deployment

### PR DESCRIPTION
**Pull request by accorvin**

### CODE REVIEW

File: rsyslog/overlays/prod/secrets/kafka-cert.enc.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+apiVersion: ENC[AES256_GCM,data:EFo=,iv:gh0/qFGqpgoU0AiNCuinhIafJ6JjZK+yxRSAuiHZ8m0=,tag:pKSxf3ui/bVlf8wrjOoq5Q==,type:str]
+kind: ENC[AES256_GCM,data:A0TfR7IJ,iv:hs1ra3yBeEJ5bGmKyVFWwF+QjKIRzhJjkENmHXpR+D4=,tag:tuairdDHTH8YYsJqtVuIjg==,type:str]
+metadata:
+    name: ENC[AES256_GCM,data:0HBDQKrW20dVsw==,iv:C+7yp8fCPRsl0p6C5NYyVPyDjhrrHyYG8afUla6qx2o=,tag:FV0YmcU0+6M5sFjJR7I79A==,type:str]
```
Comment (anishasthana) [inline rsyslog/overlays/prod/secrets/kafka-cert.enc.yaml]: Does the name really need to be encrypted?

Comment (accorvin) [reply rsyslog/overlays/prod/secrets/kafka-cert.enc.yaml]: This was just the output of `sops -e overlays/prod/secrets/kafka-cert.yaml > overlays/prod/secrets/kafka-cert.enc.yaml`.

Is there a way to not have those fields encrypted?

File: rsyslog/bases/rsyslog/rsyslog.yaml [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+      protocol: UDP
+      targetPort: 10514
+  selector:
+    app: rsyslog
+    rsyslog-name: ingestor
+  sessionAffinity: None
+  type: NodePort
+
+---
+ apiVersion: v1
```
Comment (HumairAK) [inline rsyslog/bases/rsyslog/rsyslog.yaml:129]: There seems to be additional indents on this configmap

File: rsyslog/bases/rsyslog/rsyslog.yaml [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+ kind: ConfigMap
+ metadata:
+   name: rsyslog-d-conf
+ data:
```
Comment (HumairAK) [inline rsyslog/bases/rsyslog/rsyslog.yaml:133]: If you are willing -- I would suggest using a [configmap generator](https://kubectl.docs.kubernetes.io/pages/reference/kustomize.html#configmapgenerator) here. Like how we use it for jupyterhub [here](https://github.com/AICoE/internal-data-hub/blob/production/jupyterhub/bases/jupyterhub/kustomization.yaml#L9).

**Comment by anishasthana** [comment]
Can you reopen the PR at https://github.com/AICoE/internal-data-hub? Thanks!

File: rsyslog/bases/rsyslog/rsyslog.yaml (hunk 2 above)
Comment (accorvin) [reply rsyslog/bases/rsyslog/rsyslog.yaml:129]: Fixed in new PR

File: rsyslog/bases/rsyslog/rsyslog.yaml (hunk 3 above)
Comment (accorvin) [reply rsyslog/bases/rsyslog/rsyslog.yaml:133]: Fixed in new PR

**Comment by accorvin** [comment]
> Can you reopen the PR at https://github.com/AICoE/internal-data-hub? Thanks!

Done. See https://github.com/AICoE/internal-data-hub/pull/1. Closing this one.

**Pull request closed without merging**